influxql/engine_test.go

```diff
@@ -0,0 +1,383 @@
+package influxql
+
+import (
+	"fmt"
+	"testing"
+	"time"
+)
+
+func derivativeJob(t *testing.T, fn, interval string) *MapReduceJob {
+	q, err := ParseQuery(fmt.Sprintf("SELECT %s(mean(value), %s) FROM foo", fn, interval))
+	if err != nil {
+		t.Fatalf("failed to parse query: %s", err)
+	}
+	m := &MapReduceJob{
+		stmt: q.Statements[0].(*SelectStatement),
+	}
+
+	return m
+}
+
+func TestProcessDerivative(t *testing.T) {
```

Comment:
I would find GoDoc strings here very useful so I quickly know what to expect from the test.